otpilot: Make spi_processor less noisy

osenft opened the pull request:
This change adds explicit handling for SPI command 0xC8 (read extended address register). It further adds an option for conditionally enabling the SpiProcessor to print the headers of incoming SPI packets, defaulting to "off" (it was previously always printing these headers, causing a lot of noise on the console).

dann21 commented on userspace/otpilot/src/spi_processor.rs:
```diff
@@ -269,6 +283,11 @@ impl<'a> SpiProcessor<'a> {
+            OpCode::ReadExtendedAddressRegister => {
+                // We can't handle this, but we also don't want it to raise an
+                // UnsupportOpCode error.
+                self.ignore_transaction()
```
Would it make sense to spew something here, so that an unsuspecting victim will see evidence that the command they sent is getting ignored?

osenft replied:
You could make the argument that we generally want to print information about unsupported commands. I found this particular one by testing against u-boot running on Nuvoton. I'm not why it's making this call since the extended address register is not supported in Haven. Looks like a bug in the Nuvoton u-boot code to me, really. Since it's not supported, it'll return `0xff` on SPI. I believe that works exactly the same as if this was attempted with a regular SPI flash chip that doesn't support the command.

osenft replied:
With that, I think you're right, we should not just silently ignore it. I removed the special handling code for the `ReadExtendedAddressRegister` again. This results in `process_spi_header()` returning an error which is then handled (by printing) in the `run()` loop. I think this is actually cleaner.

dann21 reviewed (approved):
LGTM (modulo one tentative suggestion)

osenft committed:
otpilot: Make spi_processor less noisy

This change adds adds an option for conditionally enabling the
SpiProcessor to print the headers of incoming SPI packets,
defaulting to "off" (it was previously always printing these
headers, causing a lot of noise on the console).
